pytest-embedded-idf/pytest_embedded_idf/dut.py

```diff
@@ -76,11 +91,73 @@ def toolchain_prefix(self) -> str:
         else:
             raise ValueError(f'Unknown target: {self.target}')
 
+    @property
+    def panic_output_decode_script(self) -> Optional[str]:
+        """
+        Returns:
+            Panic output decode script path
+        """
+        script_filepath = self._panic_output_decode_script or os.path.join(
+            os.getenv('IDF_PATH', 'IDF_PATH'),
+            'tools',
+            'gdb_panic_server.py',
+        )
+        if not os.path.isfile(script_filepath):
+            logging.info(
+                'Panic output decode script not found. Please use --panic-output-decode-script flag '
+                'to provide script or set IDF_PATH (Default: $IDF_PATH/tools/gdb_panic_server.py)'
+            )
+            raise ValueError
+        return os.path.realpath(script_filepath)
+
+    def _check_panic_decode_trigger(self):  # type: () -> None
+        with open(self.pexpect_proc._fr.name, 'rb') as output_file:
+            output = output_file.read()
+        panic_output_res = self.PANIC_OUTPUT_REGEX.search(output)
```

Comment:
large regex string often would cost a longer time for search. imaging such a use case.

A test script is expecting with a timeout as 30 seconds. The app run into core dump, which would make the app restart over and over again, for 30 seconds. The output of the file would be extremely large.

get the first "start line" and the first "end line", parse the string between them would be better.